mtp_cashbook/apps/cashbook/views.py

```diff
@@ -1,27 +1,47 @@
 from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+from django.core.urlresolvers import reverse_lazy
 from django.utils.decorators import method_decorator
-from django.views.generic import ListView
-from mtp_cashbook.apps.mtp_auth.api_client import get_connection
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import FormView
 
+from .forms import ProcessTransactionBatchForm
 
-class TransactionBatchListView(ListView):
 
+class TransactionBatchListView(FormView):
+
+    form_class = ProcessTransactionBatchForm
     template_name = 'cashbook/transaction_batch_list.html'
+    success_url = reverse_lazy('index')
+
+    def get_form_kwargs(self):
+        form_kwargs = super(TransactionBatchListView, self).get_form_kwargs()
+        form_kwargs['request'] = self.request
+        return form_kwargs
 
     @method_decorator(login_required)
     def dispatch(self, request, *args, **kwargs):
-        self.client = get_connection(request)
         return super(TransactionBatchListView, self).dispatch(request, *args, **kwargs)
 
-    def get_queryset(self):
-        user = self.request.user
-
-        resp = self.client.transactions(user.prison)(user.pk).get(status='pending')
-        if resp.get('count') == 0:
-            resp = self.client.transactions(user.prison)(user.pk).take.post()
-        return resp.get('results', [])
-
     def get_context_data(self, **kwargs):
         context = super(TransactionBatchListView, self).get_context_data(**kwargs)
-        context['total'] = sum([x['amount'] for x in context['object_list']])
+
+        transaction_choices = context['form'].transaction_choices
+        context['object_list'] = transaction_choices
+        context['total'] = sum([x[1]['amount'] for x in transaction_choices])
         return context
+
+    def form_valid(self, form):
+        credited, discarded = form.save()
+
+        messages.success(
+            self.request,
+            _(
+                'Batch processed successfully. \
+                %(credited)s credited, %(discarded)s discarded.'
```

Comment:
Hm. We use `%()` here and `{{}}` in templates. I wonder if we can make those consistent to make copy-editing easier?